File: backend/services/cover_letter_service.py
from services.gemini_service import model


def generate_cover_letter(
    resume_text,
    job_description
):

    prompt = f"""
    Generate a professional cover letter.

    Resume:
    {resume_text}

    Job Description:
    {job_description}

    Requirements:
    - Professional tone
    - 300-500 words
    - Highlight relevant skills
    - Explain why the candidate is suitable
    """

    response = model.generate_content(
        prompt
    )

    return {
        "cover_letter":
            response.text
    }

File: frontend/services/coverLetterService.js
import api from "./api";

export const generateCoverLetter =
  async (
    resumeText,
    jobDescription
  ) => {

    const token =
      localStorage.getItem("token");

    const response =
      await api.post(
        "/generate-cover-letter",
        {
          resume_text: resumeText,
          job_description:
            jobDescription
        },
        {
          headers: {
            Authorization:
              `Bearer ${token}`
          }
        }
      );

    return response.data;
};